### DISCUSSION TITLE
additional variables and default values in Route

### DISCUSSION THREAD

**Discussion by sandeep-gh** [encode/starlette#1705] [Q&A] [answered]
It seems in Flask, we can specify routes as follows:
```
@frontend.route('/<itemname:item_name>', defaults=dict(rev=CURRENT), methods=['GET', 'POST'])
```
which essentially injects defaults in the function's namespace. Was wondering if there something similar available for Starlette or FastAPI. Should we use partial function bindings?

**Comment by Kludex** [comment, marked as answer]
There's no such a thing for Starlette.
